types/module/server/module.go

```diff
@@ -0,0 +1,20 @@
+package server
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdkmodule "github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/regen-network/regen-ledger/types/module"
+)
+
+type Module interface {
+	module.ModuleBase
+
+	RegisterServices(Configurator)
+}
+
+type Configurator interface {
+	sdkmodule.Configurator
+
+	ModuleKey() ModuleKey
+	BinaryMarshaler() codec.BinaryMarshaler
```

Comment:
don't we also need `RequireServer(serverInterface interface{})`?

Reply:
Yeah we do but in the simplest PoC we can just fail at runtime

Reply:
ended up adding it anyway